src/marqo/s2_inference/multimodal_utils.py

```
@@ -0,0 +1,286 @@
+            buffer.write(chunk)
+            if buffer.tell() >= sample_size:
+                break
+        buffer.seek(0)
+        yield buffer
+    finally:
+        buffer.close()
+        response.close()
+
+def infer_modality(content: Union[str, List[str], bytes]) -> Modality:
+    if isinstance(content, str):
+        if not validators.url(content):
+            return Modality.TEXT
+        
+        extension = content.split('.')[-1].lower()
+        if extension in ['jpg', 'jpeg', 'png', 'gif']:
+            return Modality.IMAGE
+        elif extension in ['mp4', 'avi', 'mov']:
+            return Modality.VIDEO
+        elif extension in ['mp3', 'wav', 'ogg']:
+            return Modality.AUDIO
+        
+        if validate_url(content):
+            # Use context manager to handle content sample
+            try:
+                with fetch_content_sample(content) as sample:
+                    #print(f"did not find extension, but found url: {content}")
+                    mime = magic.from_buffer(sample.read(), mime=True)
+                    #print(f"mime: {mime}")
+                    if mime.startswith('image/'):
+                        return Modality.IMAGE
+                    elif mime.startswith('video/'):
+                        return Modality.VIDEO
+                    elif mime.startswith('audio/'):
+                        return Modality.AUDIO
+            except Exception as e:
+                pass
+        
+        return Modality.TEXT
+    
+    elif isinstance(content, bytes):
+        # Use python-magic for byte content
+        mime = magic.from_buffer(content, mime=True)
+        if mime.startswith('image/'):
+            return Modality.IMAGE
+        elif mime.startswith('video/'):
+            return Modality.VIDEO
+        elif mime.startswith('audio/'):
+            return Modality.AUDIO
+        else:
+            return Modality.TEXT
+    
+    return None
```

Return type isn't optional. Also how is the caller supposed to interpret None? Maybe you need to raise an exception?

---

Right, let's just raise an error. If the content is not a string, a list of strings, or bytes, there's probably something wrong

---

But what if it's a number? Should be treated as text right? I will return Text for now